src/axom/core/memory_management.hpp

```diff
@@ -191,7 +193,7 @@ inline void deallocate( T*& pointer ) noexcept
 
 //------------------------------------------------------------------------------
 template < typename T >
-inline T* reallocate( T* pointer, std::size_t n ) noexcept
+inline T* reallocate( T* pointer, std::size_t n, int allocID ) noexcept
```

Comment:
Hmm...I am not sure what it means to reallocate a pointer using a different allocator (?) 

When we reallocate a pointer, umpire can give us the allocator ID that was used to reallocate the pointer. Does `reallocate` makes sense on a different allocator ID?

@davidbeckingsale , @rhornung67 -- Any thoughts on that?

Reply:
For the case of a null pointer, `axom::reallocate()` will use the allocator associated with the given allocator ID (or default allocator if not provided) to allocate the data.
I was trying to make the usage similar to [umpire:: reallocate() ](https://github.com/LLNL/Umpire/blob/817779995749109b09334bed5a4b7a9ce1bbe166/src/umpire/ResourceManager.hpp#L227).

Reply:
I see -- You are absolutely right.

Perhaps adding a `\note` in the doxygen documentation indicating that the allocator ID is only used when a null pointer is supplied and otherwise ignored would be helpful.